### PR TITLE
Uses tooltip to show custom popups for mute mic button.

### PR DESCRIPTION
**Pull request by damencho**
Updates popups shown when auto-muted mic or shared video, or the popups shown when talk while muted is detected. Using the standard tooltips now.

Before:
b1
b2
b3

After:
a1
a2
a3

### CODE REVIEW

File: modules/UI/toolbars/Toolbar.js [hunk 1]
```diff
@@ -716,25 +720,55 @@ Toolbar = {
...
-            popupElement.id = popup.id;
-            popupElement.className = popup.className;
-            let liElement = document.createElement("li");
-            liElement.setAttribute("data-i18n", popup.dataAttr);
-            popupElement.appendChild(liElement);
-            buttonElement.appendChild(popupElement);
-        });
+    /**
+     * Show custom popup/tooltip for a specified button.
+     */
```
Comment (yanas) [inline modules/UI/toolbars/Toolbar.js]: Missing @param js doc

**Comment by yanas** [comment]
Something weird is happening with the timeout of the custom tooltip. It sometimes disappears too quickly or sometime it doesn't show up (for example the tooltip saying that we've been automatically muted during the shared video doesn't always appear). Some of the bugs I noticed may be related to the shared video code and not be regressions.

File: modules/UI/toolbars/Toolbar.js [hunk 2]
```diff
@@ -263,11 +263,14 @@ const defaultToolbarButtons = {
...
         tooltipKey: 'toolbar.sharedvideo',
         className: 'button icon-shared-video',
-        html: `<ul id="sharedVideoMutedPopup" 
-                   class="loginmenu extendedToolbarPopup">
-                   <li data-i18n="[html]toolbar.sharedVideoMutedPopup"></li>
-               </ul>
-`
+        popups: [
+            {
+                id: "sharedVideoMutedPopup",
```
Comment (saghul) [inline modules/UI/toolbars/Toolbar.js]: single quotes?

File: modules/UI/toolbars/Toolbar.js [hunk 3]
```diff
@@ -724,17 +733,47 @@ Toolbar = {
 
     _addPopups(buttonElement, popups = []) {
         popups.forEach((popup) => {
-            let popupElement = document.createElement("ul");
+            let popupElement = document.createElement("div");
```
Comment (saghul) [inline modules/UI/toolbars/Toolbar.js]: since you're here, make it const?

File: modules/UI/toolbars/Toolbar.js [hunk 4]
```diff
@@ -724,17 +733,47 @@ Toolbar = {
...
 
     /**
+     * Show custom popup/tooltip for a specified button.
+     * @param popupSelectorID the selector id of the popup to show
+     * @param show true or false/show or hide the popup
+     * @param timeout the time to show the popup
+     */
+    _showCustomToolbarPopup(popupSelectorID, show, timeout) {
+
+        let gravity = $(popupSelectorID).attr('data-tooltip');
```
Comment (saghul) [inline modules/UI/toolbars/Toolbar.js]: const

File: modules/UI/toolbars/Toolbar.js [hunk 5]
```diff
@@ -724,17 +733,47 @@ Toolbar = {
...
+        AJS.$(popupSelectorID)
+            .tooltip({
+                trigger: 'manual',
+                html: true,
+                gravity: gravity,
+                title: 'title'});
+        if (show) {
+            AJS.$(popupSelectorID).tooltip("show");
+            setTimeout(function () {
```
Comment (saghul) [inline modules/UI/toolbars/Toolbar.js]: arrow function?

**Comment by saghul** [comment]
Hey @damencho, left a few general comments.

**Comment by damencho** [comment]
Hey @saghul I just addressed your comments.

**Pull request merged**